server/sonar-server/src/test/java/org/sonar/server/computation/task/projectanalysis/webhook/WebhookPayloadTest.java

```diff
@@ -94,14 +86,52 @@ public void create_payload_with_gate_conditions_without_value() {
         .setErrorThreshold("70.0")
         .buildNoValue())
       .build();
-    PostProjectAnalysisTask.ProjectAnalysis analysis = newAnalysis(task, gate);
+    PostProjectAnalysisTask.ProjectAnalysis analysis = newAnalysis(task, gate, emptyMap());
 
     WebhookPayload payload = WebhookPayload.from(analysis);
     assertThat(payload.getProjectKey()).isEqualTo(PROJECT_KEY);
     assertJson(payload.toJson()).isSimilarTo(getClass().getResource("WebhookPayloadTest/gate_condition_without_value.json"));
   }
 
-  private static PostProjectAnalysisTask.ProjectAnalysis newAnalysis(CeTask task, @Nullable QualityGate gate) {
+  @Test
+  public void create_payload_with_analysis_properties() {
+    CeTask task = newCeTaskBuilder()
+      .setStatus(CeTask.Status.SUCCESS)
+      .setId("#1")
+      .build();
+    QualityGate gate = newQualityGateBuilder()
+      .setId("G1")
+      .setName("Gate One")
+      .setStatus(QualityGate.Status.WARN)
+      .build();
+    Map<String, String> scannerProperties = ImmutableMap.of(
+      "sonar.analysis.revision", "ab45d24",
+      "sonar.analysis.buildNumber", "B123",
+      "not.prefixed.with.sonar.analysis", "should be ignored",
+      "foo", "should be ignored too"
+    );
+    PostProjectAnalysisTask.ProjectAnalysis analysis = newAnalysis(task, gate, scannerProperties);
+
+    WebhookPayload payload = WebhookPayload.from(analysis);
+    assertJson(payload.toJson()).isSimilarTo(getClass().getResource("WebhookPayloadTest/with_analysis_properties.json"));
+    assertThat(payload.toJson())
+      .doesNotContain("not.prefixed.with.sonar.analysis")
+      .doesNotContain("foo");
```

Comment:
you could test for "ignored" in case the property key is incorrect but value is incorrectly added